apps/admin-gui/src/app/shared/components/dialogs/update-application-form-dialog/update-application-form-dialog.component.ts

```diff
@@ -34,21 +35,50 @@ export class UpdateApplicationFormDialogComponent implements OnInit {
   ngOnInit(): void {
     this.theme = this.data.theme;
     this.applicationForm = this.data.applicationForm;
-    this.moduleName = this.applicationForm.moduleClassName;
+    for (const moduleName of this.applicationForm.moduleClassNames) {
+      this.addModule(moduleName);
+    }
+    if (this.formControls.length === 0) {
+      this.addModule();
+    }
     this.initialState = this.applicationForm.automaticApproval ? 'auto' : 'manual';
     this.extensionState = this.applicationForm.automaticApprovalExtension ? 'auto' : 'manual';
     this.embeddedState = this.applicationForm.automaticApprovalEmbedded ? 'auto' : 'manual';
     this.entity = this.data.entity;
     this.autoRegistrationEnabled = this.data.autoRegistrationEnabled;
   }
 
+  addModule(name = ''): void {
+    const commaControl = new FormControl(name, {
+      validators: [Validators.pattern(/^[^,]*$/)],
+      updateOn: 'change',
+    });
+    commaControl.markAsTouched(); // This helps show the mat-error message immediately when a comma is typed
+    this.formControls.push(commaControl);
+  }
+
+  removeModule(index: number): void {
+    this.formControls.splice(index, 1);
+  }
+
+  trackByIndex(index: number): number {
+    return index;
+  }
+
+  formValid(): boolean {
+    return this.formControls.every((control) => control.valid);
```

Comment:
I believe a more "correct" solution would be to create a FormGroup and add a new control for each module. This way you could also just use FormGroup.invalid (when disabling the confirmation button )in the template instead of a function call (which we try to avoid since it's inefficient). See https://angular.io/api/forms/FormGroup and components like password-reset-form.